### PR TITLE
MM-42303 Change modal-header CSS to use flexbox instead of floats

### PR DESCRIPTION
**Pull request by hmhealey**
This ticket is to fix an issue where long channel names caused the Manage Channel Members modal title to look funny. I also ended up fixing an issue where the Add Members button in that header wasn't visible at all in mobile web view.

This was a bit more of a rabbit hole than I was expecting to go down today, but I think it ended up turning out alright. Previously, we applied `float: left` to the header text and `float: right` to the buttons that sometimes appear there which makes positioning everything miserable. Now, we just use flexbox like everyone who doesn't have to support IE11 any more should.

Note that the close button in the modal header is still using a float, which is why I increased the right padding on the `modal-header` class, but that's part of our ancient version of Bootstrap, so I didn't want to mess with it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42303

#### Screenshots
Screen Shot 2022-03-08 at 2 54 21 PM
Screen Shot 2022-03-08 at 2 54 29 PM

#### Release Note
```release-note
Fixed overflow of text in Manage Channel Members modal title
```

### CODE REVIEW

File: sass/components/_modal.scss [hunk 1]
```diff
@@ -240,8 +240,9 @@
     }
 
     .modal-header {
+        display: flex;
         min-height: 56px;
-        padding: 14px 15px 11px;
+        padding: 14px 48px 11px 15px;
```
Comment (michelengelen) [inline sass/components/_modal.scss:245]: whats that massive padding-right doing? And why is it begin reset on mobile?

Is this for spacing out the button? If yes, this could be done better since this will break on translations with longer texts in the buttons.

File: sass/components/_modal.scss [hunk 2]
```diff
@@ -253,8 +254,8 @@
         .modal-title {
             overflow: hidden;
             max-width: calc(100% - 80px);
+            flex-grow: 1;
```
Comment (michelengelen) [inline sass/components/_modal.scss:257]: nit: you could just use `flex: 1` here

File: sass/components/_modal.scss [hunk 3]
```diff
@@ -315,8 +316,7 @@
         .btn {
             &.btn-primary {
                 position: relative;
-                margin: -4px 33px 0 0;
-                float: right;
+                margin-top: -4px;
```
Comment (michelengelen) [inline sass/components/_modal.scss:319]: aligning items inside flex containers would make these tiny adjustments irrelevant ... maybe try `align-items: center` on the container?

File: sass/responsive/_mobile.scss [hunk 4]
```diff
@@ -375,20 +375,25 @@
...
                         display: block;
                         width: 100%;
                         margin: 10px 0 6px;
-                        float: none;
                     }
                 }
             }
         }
 
+        .more-modal {
```
Comment (michelengelen) [inline sass/responsive/_mobile.scss:390]: definitely not part of this PR, but I always thought that these very specific changes should be done inside the components SCSS

**Review by michelengelen — CHANGES_REQUESTED** [review]
only small suggestions, otherwise looks pretty good

**Review by larkox — APPROVED** [review]

File: sass/components/_modal.scss (hunk 1 above)
Comment (hmhealey) [reply sass/components/_modal.scss:245]: It's so that the button/title doesn't overlap the close button in desktop view since the close button still has `float: right` applied to it. Previously, that padding was actually a margin on the button (you can see the old margin above [this comment](https://github.com/mattermost/mattermost-webapp/pull/9975#discussion_r822379007) but it only applied if there was a non-close button in the modal header.

The reason it's not needed in mobile view is because the rest of the header styling is different there. If you check out the screenshots above, the text and close button are on a separate line from the add button. When I had that padding in mobile view, the button wouldn't stretch across the full width available.

File: sass/components/_modal.scss (hunk 3 above)
Comment (hmhealey) [reply sass/components/_modal.scss:319]: I tried that, but it's still out of place vertically. The `.modal-header` (as seen above) has uneven top and bottom padding, so that's probably the cause of it, but I'm worried that changing that will introduce issues in other modals because of how many places that is used. I'd rather leave this part as-is for now since it's not part of this bug

File: sass/responsive/_mobile.scss (hunk 4 above)
Comment (hmhealey) [reply sass/responsive/_mobile.scss:390]: Agreed. The modals are a mess of interconnected and miserable CSS, some of which is from our old version of Bootstrap and some that's specifically to revert the changes from Bootstrap. This is definitely one of those areas that needs a lot of rework or just to be abandoned next time we start reworking modals again.

File: sass/components/_modal.scss (hunk 1 above)
Comment (michelengelen) [reply sass/components/_modal.scss:245]: just as i thought ... still pretty hacky! :D

**Review by michelengelen — APPROVED** [review]
LGTM

File: sass/components/_modal.scss (hunk 1 above)
Comment (hmhealey) [reply sass/components/_modal.scss:245]: Oh yeah, 100%. Unfortunately, we're still bound by our ancient version of Bootstrap

**Comment by jgilliam17** [comment]
Thanks @hmhealey 
Tested, looks good to merge - checked [E2E report](https://automation-dashboard.vercel.app/cycles/cd19f084-20ec-4ba7-b916-ca358923dda7), no PR related failures 
- Verified Add members button in mobile and standard view, using short and long channel names - as expected.
- Noticed another byproduct of long channel name in mobile view, filed separately out of scope here.
https://mattermost.atlassian.net/browse/MM-42752

**Review by jgilliam17 — APPROVED** [review]
